bokeh/core/properties.py

```diff
@@ -1559,6 +1559,15 @@ def prepare_value(self, cls, name, value):
             value = dict(value=value)
         return super(FontSizeSpec, self).prepare_value(cls, name, value)
 
+    def validate(self, value):
+        super(FontSizeSpec, self).validate(value)
+
+        if isinstance(value, dict) and 'value' in value:
+            value = value['value']
+
+        if isinstance(value, string_types) and value[0].isdigit() and self._font_size_re.match(value) is None:
```

Comment:
Also I'm not sure what's the need for that clause (an optimization?). The regex tests for the first character to be a digit.